dataset/src/main/kotlin/com/cosmotech/dataset/service/DatasetServiceImpl.kt

```diff
@@ -474,6 +485,24 @@ class DatasetServiceImpl(
       hasChanged = true
     }
 
+    if (dataset.security != null && existingDataset.security == null) {
```

Comment:
remove and prevent modification of security. security modification should be done in endpoint security